.github/workflows/python.yml

```diff
@@ -10,19 +10,18 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v4
-    # NB: v1.4.0 covers Python 3.8.
-    - uses: ricardochaves/python-lint@v1.4.0
+    - uses: actions/setup-python@v5
       with:
-        python-root-list: lddtree.py pylint
-        use-pylint: true
-        use-pycodestyle: false
-        use-flake8: false
-        use-black: true
-        use-mypy: true
-        use-isort: true
-        extra-pylint-options: ""
-        extra-pycodestyle-options: ""
-        extra-flake8-options: ""
-        extra-black-options: ""
-        extra-mypy-options: ""
-        extra-isort-options: ""
+        python-version: '3.x'
+    - name: install dependencies
+      run: |
+        python -m pip install pylint mypy black 'isort[colors]'
+        # mypy dependencies for following imported types
+        python -m pip install argcomplete
+    - run: pylint --output-format colorized lddtree.py
```

Comment:
we should hoist the list of paths to check to a variable so we only have to list it once